### ISSUE TITLE
Card keyboard navigation

### ISSUE THREAD

**Issue by evansjohnson** [palantir/blueprint#6870]
#### Steps to reproduce
https://blueprintjs.com/docs/#core/components/card

- Toggle on "interactive" prop
- Attempt to navigate to with keyboard

#### Actual behavior
Cannot use keyboard to navigate to and interact with an interactive `Card` element

#### Expected behavior
Possibly should only be keyboard accessible if `onClick` is defined, but if so, should be able to trigger with keyboard.

#### Possible solution
Use a generic wrapper component that hooks up keyboard navigation support when adding `onClick` to elements that are not keyboard accessible by default.

Similar issue to https://github.com/palantir/blueprint/issues/4782, but in this issue cannot navigate to with keyboard at all, in addition to not being able to trigger action with keyboard.

**Comment by evansjohnson** [comment]
Also applies to the related [CardList](https://blueprintjs.com/docs/#core/components/card-list) component
